service/authentication/src/authenticationservice.cpp

```diff
@@ -0,0 +1,83 @@
+#include <util/base64.h>
+
+#include <authenticationservice/authenticationservice.h>
+
+using namespace cc::webserver;
+
+namespace cc
+{
+namespace service
+{
+namespace authentication
+{
+
+AuthenticationServiceHandler::AuthenticationServiceHandler(
+  const ServerContext& ctx_)
+  : _sessionManager(ctx_.sessionManager)
+{
+}
+
+bool AuthenticationServiceHandler::isRequiringAuthentication()
+{
+  return _sessionManager.auth_isRequiringAuthentication();
+}
+
+bool AuthenticationServiceHandler::isCurrentSessionValid()
+{
+  return _sessionManager.auth_isCurrentSessionValid();
+}
+
+void AuthenticationServiceHandler::getAuthPrompt(std::string& ret_)
+{
+  ret_ = _sessionManager.auth_getAuthPrompt();
+}
+
+void AuthenticationServiceHandler::getAcceptedAuthMethods(
```

Comment:
There are **two** authentication methods in CodeChecker, the one with username and password, and the one with the pre-generated token.
While I wasn't there when that feature was implemented, I believe the method of using the pre-generated token instead of a password is more of a hack. It happened as it is, but in case CodeCompass is to support token-based authentication, or OAuth, this extra round will be needed.